hasher.go

```diff
@@ -31,10 +40,33 @@ type ValueHasher interface {
 	ValueHashSize() int
 }
 
+// bufferPool provides reusable byte slices to reduce allocations
+var bufferPool = sync.Pool{
+	New: func() any {
+		// Pre-allocate reasonable capacity
+		return make([]byte, 0, bufferPoolPreallocationSize)
+	},
+}
+
+// getBuffer returns a reusable byte slice from the pool
+func getBuffer() []byte {
+	return bufferPool.Get().([]byte)[:0] // Reset length but keep capacity
+}
+
+// putBuffer returns a byte slice to the pool for reuse
+func putBuffer(buf []byte) {
+	// Don't pool very large buffers
+	if cap(buf) < maxSizeForBufferPool {
+		bufferPool.Put(buf)
```

Comment:
We should fix or disable this linter check:
```
SA6002: argument should be pointer-like to avoid allocations (staticcheck)
```

Reply:
[`923465c` (#53)](https://github.com/pokt-network/smt/pull/53/commits/923465c7ab9311dadbe307e8bc59c025e2d54554)